deep_qa/data/instances/mc_tuples_with_background_instance.py

```diff
@@ -0,0 +1,358 @@
+                           curr_answer_tuples in self.answer_tuples]
+        background_indexed = self.index_tuples(self.background_tuples, data_indexer)
+        return IndexedMCTuplesWithBackgroundInstance(answers_indexed,
+                                                     background_indexed,
+                                                     self.label,
+                                                     self.index)
+
+    # This is a helped method for to_indexed_instance which indexes a list of TextTuples
+    # Input:
+    #   tuples_in is a list of text tuples, each tuple is converted to a list of strings, then
+    #   each of the strings is converted to a list of indices.
+    # Return:
+    #   indexed: List[List[List[int]]]
+    def index_tuples(self, tuples_in: List[TextTuple], data_indexer: DataIndexer):
+        indexed = []
+        for text_tuple in tuples_in:
+            text_list = text_tuple.to_text_list(self.object_handling, self.use_context)
+            # indices: List[List[int]]
+            indices = [self._index_text(phrase, data_indexer) for phrase in text_list]
+            indexed.append(indices)
+        return indexed
+
+    @classmethod
+    def read_from_line(cls, line: str, default_label: int=None):
+        """
+        Reads a TupleInstances from a line.  The format has one option (currently):
+            [question index][tab][num answer coptions][tab][all question+answer tuples][tab]
+            [background tuples]
+        The question+answer tuples are formatted as:
+            [a_1-tuple_1]$$$...$$$[a_1-tuple_n]###...###[a+_m-tuple_1]$$$...$$$||[a_m-tuple_p]
+        such that ``$$$`` serves as the tuple delimiter and ``###`` serves as the answer candidate
+        delimiter.
+        Each of the tuples is formatted as:
+            [s words]|[v words]|[o_1 words]|...|[o_j words]|[context: context words]
+        such that ``|`` serves as the tuple-slot delimiter.  Note that some slots may contain prefixes
+        indicating the type of information they contain (e.g., "context:", see below for more).
+
+        Objects are optional, and can vary in number. This makes the acceptable number of slots per
+        tuple 2 or more.
+
+        If a tuple slot is reserved for context, it contains "context:" at the start.
+
+        Some tuples contain coreference resolution info (from one of the table sources).  We do not
+        currently handle this, but plan to eventually replace the referring expression with its resolved
+        value.
+            Example: it[sky]|is|blue --> would become --> sky|is|blue
+
+        Some tuples additionally contain prefixed information about some of the "objects" such as "L:" for
+        locations and T: for time.  At present this information is simply filtered out, but can be used
+        later.
+        """
+        fields = line.split("\t")
+        assert len(fields) == 5, \
```

Comment:
i think this is better written as https://github.com/allenai/deep_qa/blob/bbeb31a3749885824f0a0d9efbb8a7f6a2758c06/deep_qa/data/instances/character_span_instance.py#L59. Does the index _have_ to be there?